### PR TITLE
Ensure hidden attribute works.

### LINKED ISSUE #397: hidden attribute does not work for paper-input inside shadow dom
### Description
`paper-input` does not respect the hidden attribute when used inside a shadow root.  This is a common issue in many Polymer elements, I'm just randomly starting with `paper-input`.

### Expected outcome
The `hidden` attribute should cause `display:none` to apply.

### Actual outcome
The `:host` selector takes over, causing `paper-input` to remain visible even with hidden attribute.

### Live Demo
https://jsbin.com/gukazaxenu/edit?html,output

### Proposed fix
We should be using `:host(:not([hidden]))` selector to specify the `display` option of custom elements.

### PR DESCRIPTION
**Pull request by coreyfarrell**
Fixes #397.

### CODE REVIEW

File: demo/ssn-input.html [hunk 1]
```diff
@@ -20,6 +20,10 @@
         display: inline-block;
       }
 
+      :host([hidden]) {
+        display: none!important;
```
Comment (notwaldorf) [inline demo/ssn-input.html]: Sorry to be a nitpicker -- would you mind adding a space before the `!important`, please?

**Comment by notwaldorf** [comment]
Looks good -- just a small nit left

File: demo/ssn-input.html (hunk 1 above)
Comment (coreyfarrell) [inline demo/ssn-input.html]: No problem, I understand the value of uniform code style.  Updated commit running through CI now.

**Comment by notwaldorf** [comment]
Thanks so much! 👍

**Pull request merged**